src/compiler/crystal/types.cr

```diff
@@ -719,6 +719,26 @@ module Crystal
       (yield self) ? true : false
     end
 
+    def always_falsey?
+      remove_indirection.is_a?(NilType)
+    end
+
+    def always_truthy?
+      type = remove_indirection
+      case type
+      when NilType
+        false
+      when BoolType
+        false
+      when PointerInstanceType
+        false
```

Comment:
Could be more succinct.

```suggestion
      when NilType, BoolType, PointerInstanceType
        false
```

Reply:
I'll consider it. Sometimes I like to see things listed line by line.